src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -57,6 +57,26 @@ namespace winrt::TerminalApp::implementation
         _tabContent = this->TabContent();
         _tabRow = this->TabRow();
         _tabView = _tabRow.TabView();
+        _rearranging = false;
+
+         _tabView.TabDragStarting([this](auto&& /*o*/, auto&& /*a*/) {
+            _rearranging = true;
+            _rearrangeFrom = -1;
```

Comment:
stylistically, I'd prefer a `std::optional<>` to hold these values, and reset them to `std::nullopt` instead of `-1` to indicate they're unset.

Reply:
Good suggestion, fixed.

Reply:
Fixed